test/powershell/Modules/Microsoft.WSMan.Management/ConfigProvider.Tests.ps1

```diff
@@ -0,0 +1,478 @@
+            @{type="TEXT"}
+        ) {
+            param ($type)
+            Set-Item $testPluginPath\XmlRenderingType $type -WarningAction SilentlyContinue
+            (Get-Item $testPluginPath\XmlRenderingType).Value | Should Be $type
+        }
+
+        It "Set-Item on non-existent property should fail" {
+            { Set-Item $testPluginPath\foo "bar" } | ShouldBeErrorId "System.InvalidOperationException,Microsoft.PowerShell.Commands.SetItemCommand"
+        }
+
+        It "Set-Item on plugin Resource '<property>' property with '<value>' should succeed" -TestCases @(
+            @{property="SupportsOptions"; value=$false; expected="False"},
+            @{property="sUPPORTSoPTIONS"; value=$true; expected="true"}
+        ) {
+            param($property, $value, $expected)
+            $resource = Get-ChildItem "$testPluginPath\Resources" | Select-Object -First 1
+            Set-Item "$($resource.PSPath)\$property" $value -WarningAction SilentlyContinue
+            (Get-Item "$($resource.PSPath)\$property").Value | Should Be $expected
+        }
+
+        It "Set-Item on plugin Security Resource '<property>' property with '<value>' should succeed" -TestCases @(
+            @{property="Sddl"; value="O:NSG:BAD:P(A;;GA;;;BA)"},
+            @{property="SDDL"; value="O:NSG:BAD:P(A;;GA;;;BA)(A;;GA;;;IU)(A;;GA;;;RM)S:P(AU;FA;GA;;;WD)(AU;SA;GXGW;;;WD)"}
+        ) {
+            param($property, $value, $expected)
+            $resource = Get-ChildItem "$testPluginPath\Resources" | Select-Object -First 1
+            $security = Get-ChildItem "$($resource.PSPath)\Security" | Select-Object -First 1
+            Set-Item "$($security.PSPath)\$property" $value -WarningAction SilentlyContinue -Force
+            (Get-Item "$($security.PSPath)\$property").Value | Should Be $value
+        }
+
+        It "Set-Item on plugin Security Resource '<property>' property with invalid '<value>' should fail" -TestCases @(
+            @{property="Sddl"; value="foo"},
+            @{property="sDDL"; value="D:P(A;;GA;;;BA)"} # owner missing
+        ) {
+            param($property, $value, $expected)
+            $resource = Get-ChildItem "$testPluginPath\Resources" | Select-Object -First 1
+            $security = Get-ChildItem "$($resource.PSPath)\Security" | Select-Object -First 1
+            { Set-Item "$($security.PSPath)\$property" $value -Force } | ShouldBeErrorId "System.InvalidOperationException,Microsoft.PowerShell.Commands.SetItemCommand"
+        }
+
+        It "Set-Item on plugin InitializationParameters '<property>' property with '<value>' should succeed" -TestCases @(
+            @{property="PSVersion"; value="6.0.0"},
+            @{property="psvERSION"; value="5.1"}
+        ) {
+            param($property, $value)
+            Set-Item "$testPluginPath\InitializationParameters\$property" $value -WarningAction SilentlyContinue
+            (Get-Item "$testPluginPath\InitializationParameters\$property").Value | Should Be $value
```

Comment:
I thought about it while writing the tests, but there's enough differences in the properties and validation I think it would be overly complicated to combine them